src/asyncapidiff.ts

```diff
@@ -1,44 +1,74 @@
-import { Output, DiffOutputItem } from './types';
+import { Output, OutputType, JSONOutput, Changes, AsyncAPIDiffOptions } from './types';
 import { breaking, nonBreaking, unclassified } from './constants';
+import convertToYAML from './helpers/output/convertToYAML';
 
 /**
  * Implements methods to deal with diff output.
  * @class
  * @returns {AsyncAPIDiff}
  */
 export default class AsyncAPIDiff {
-  private output: Output;
+  private output: JSONOutput;
+  private outputType: OutputType;
 
-  constructor(output: string) {
+  constructor(output: string, options: AsyncAPIDiffOptions) {
     // output is a stringified JSON
     this.output = JSON.parse(output);
+    this.outputType = options.outputType;
   }
 
   /**
-   * @returns {Array<DiffOutputItem>} All the breaking changes
+   * @returns {Changes} All the breaking changes
    */
-  breaking(): DiffOutputItem[] {
-    return this.output.changes.filter((diff) => diff.type === breaking);
+  breaking(): Changes {
+    const breakingChanges = this.output.changes.filter(
+      (diff) => diff.type === breaking
+    );
+
+    if (this.outputType === 'yaml') {
+      return convertToYAML(breakingChanges);
+    }
+
+    return breakingChanges;
```

Comment:
You can make helper function for that redundant logic (also in other functions):

```
function toProperFormat<T>(data: T, outputType: OutputType) {
  if (this.outputType === 'yaml') {
    return convertToYAML(data);
  }
  return data;
}
```

Reply:
What would be the right place for this function? In another module inside `helpers` directory or inside the `AsyncAPIDiff` class itself? :thinking:

Reply:
`helpers` will be good place :)